internal/util/secrets/secrets.go

```diff
@@ -1,55 +1,96 @@
-package util
+package secrets
 
 import (
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"log"
 	"os"
+	"os/user"
 	"path/filepath"
 	"regexp"
 
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/commitdev/commit0/internal/config"
 	"github.com/manifoldco/promptui"
 	"gopkg.in/yaml.v2"
 )
 
 // Secrets - AWS prompted credentials
 type Secrets struct {
-	Aws struct {
-		AwsAccessKeyID     string
-		AwsSecretAccessKey string
-		Region             string
-	}
+	AWS AWS
+}
+
+type AWS struct {
+	AccessKeyID     string
+	SecretAccessKey string
 }
 
-func MakeAwsEnvars(awsSecrets Secrets) []string {
+func MakeAwsEnvars(cfg *config.Commit0Config, awsSecrets Secrets) []string {
 	env := os.Environ()
-	env = append(env, fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", awsSecrets.Aws.AwsAccessKeyID))
-	env = append(env, fmt.Sprintf("AWS_SECRET_ACCESS_KEY=%s", awsSecrets.Aws.AwsSecretAccessKey))
-	env = append(env, fmt.Sprintf("AWS_DEFAULT_REGION=%s", awsSecrets.Aws.Region))
+	env = append(env, fmt.Sprintf("AWS_ACCESS_KEY_ID=%s", awsSecrets.AWS.AccessKeyID))
+	env = append(env, fmt.Sprintf("AWS_SECRET_ACCESS_KEY=%s", awsSecrets.AWS.SecretAccessKey))
+	env = append(env, fmt.Sprintf("AWS_DEFAULT_REGION=%s", cfg.Infrastructure.AWS.Region))
 
 	return env
 }
 
-func GetSecrets() Secrets {
-	dir := GetCwd()
+func GetSecrets(baseDir string) Secrets {
 
-	if fileExists(fmt.Sprintf("%s/secrets.yaml", dir)) {
+	secretsFile := filepath.Join(baseDir, "secrets.yaml")
+	if fileExists(secretsFile) {
 		log.Println("secrets.yaml exists ...")
-		return readSecrets()
+		return readSecrets(secretsFile)
 	} else {
-		awsSecrets := promptCredentials()
-		writeSecrets(awsSecrets)
-		return awsSecrets
-	}
-}
+		// Get the user's home dir
+		usr, err := user.Current()
+		if err != nil {
+			log.Fatal(err)
+		}
 
-func readSecrets() Secrets {
+		var awsSecrets Secrets
+
+		// Load the credentials file to look for profiles
+		credsFile := filepath.Join(usr.HomeDir, ".aws/credentials")
+		creds, err := ioutil.ReadFile(credsFile)
+		if err == nil {
+			// Get all profiles
+			re := regexp.MustCompile(`\[(.*)\]`)
+			profileMatches := re.FindAllStringSubmatch(string(creds), -1)
+			profiles := make([]string, len(profileMatches))
+			for i, p := range profileMatches {
+				profiles[i] = p[1]
```

Comment:
should the `1` magic number be 0 to get the `default` profile?

Reply:
nvm :man_facepalming:

Reply:
FindAllStringSubmatch returns a slice of tuples where index 0 is the full match and index 1 is the submatch